Composer/packages/lib/shared/src/schemaUtils/dereference.ts

```diff
@@ -80,6 +80,8 @@ export function dereference<S extends JSONSchema7 | JSONSchema7[] | JSONSchema7D
 export function dereferenceDefinitions(definitions: SchemaDefinitions): SchemaDefinitions {
   const resolvedDefs: SchemaDefinitions = {};
 
+  definitionCache.clear();
```

Comment:
So, based on your investigation, re-init this definitionCache will falls into an infinite loop, right? but do it once is ok, so you clear it, make sure every time deferencing start from scratch.  

that perhaps work as a hot fix, FYI @a-b-r-o-w-n, may know this better

Reply:
I think we can clear the cache when init a bot or if a type is detected in cache, we return the value directly.

Reply:
It's not very clear to me why this prevents the stackoverflow. I don't disagree with doing this though -- its reasonable that a schema definition could be different if the schema has been modified.